solidity/contracts/Chainlinked.sol

```diff
@@ -1,35 +1,64 @@
-pragma solidity ^0.4.18;
+pragma solidity ^0.4.23;
 
+import "./ChainlinkLib.sol";
+import "./LinkToken.sol";
 import "./Oracle.sol";
-import "./Chainlink.sol";
+import "./Buffer.sol";
+import "./CBOR.sol";
 
 
 contract Chainlinked {
-  using Chainlink for Chainlink.Run;
+  using ChainlinkLib for ChainlinkLib.Run;
+  using CBOR for Buffer.buffer;
 
+  uint256 constant clArgsVersion = 1;
+  bytes4 constant oracleFid = bytes4(keccak256("requestData(uint256,bytes32,address,bytes4,bytes32,bytes)"));
+
+  LinkToken internal link;
   Oracle internal oracle;
+  uint256 internal requests = 1;
 
   function newRun(
     bytes32 _jobId,
     address _callbackAddress,
     string _callbackFunctionSignature
-  ) internal returns (Chainlink.Run) {
-    Chainlink.Run memory run;
+  ) internal returns (ChainlinkLib.Run memory) {
+    ChainlinkLib.Run memory run;
+    Buffer.init(run.buf, 128);
+    run.id = keccak256(this, requests++);
```

Comment:
After being set, this could be fudged by the consuming contract. This is why it's being pulled into Oracle.sol and becoming externalId.